azure/compute/src/main/java/org/jclouds/azurecompute/compute/strategy/AzureComputeServiceAdapter.java

```diff
@@ -0,0 +1,326 @@
+    private final Function<HostedServiceGetDetailedResponse, VirtualMachine> serviceResponseToVirtualMachine;
+
+    private final Function<VirtualMachineOSImageListResponse.VirtualMachineOSImage, OSImage> osImageListResponseToOSImage;
+    private final Function<VirtualMachineOSImageGetResponse, OSImage> osImageResponseToOSImage;
+
+    private final GroupNamingConvention.Factory groupNamingConvention;
+
+    @Inject
+    protected AzureComputeServiceAdapter(AzureComputeApi client,
+                                         Function<DeploymentGetResponse, VirtualMachine> deploymentResponseToVirtualMachine,
+                                         Function<HostedServiceGetDetailedResponse, VirtualMachine> serviceResponseToVirtualMachine,
+                                         Function<VirtualMachineOSImageListResponse.VirtualMachineOSImage, OSImage> osImageListResponseToOSImage,
+                                         Function<VirtualMachineOSImageGetResponse, OSImage> osImageResponseToOSImage,
+                                         GroupNamingConvention.Factory namingConvention,
+                                         Timeouts timeouts) {
+        this.client = checkNotNull(client, "client");
+        this.deploymentResponseToVirtualMachine = checkNotNull(deploymentResponseToVirtualMachine, "deploymentResponseToVirtualMachine");
+        this.serviceResponseToVirtualMachine = checkNotNull(serviceResponseToVirtualMachine, "serviceDeploymentResponseToDeployment");
+        this.osImageListResponseToOSImage = checkNotNull(osImageListResponseToOSImage, "osImageListResponseToOSImage");
+        this.osImageResponseToOSImage = checkNotNull(osImageResponseToOSImage, "osImageResponseToOSImage");
+        this.groupNamingConvention = namingConvention;
+//      this.serverLatestJobCompleted = retry(new ServerLatestJobCompleted(client.getJobServices()),
+//            timeouts.nodeRunning * 9l / 10l);
+//      this.serverLatestJobCompletedShort = retry(new ServerLatestJobCompleted(client.getJobServices()),
+//            timeouts.nodeRunning * 1l / 10l);
+    }
+
+
+    @Override
+    public NodeAndInitialCredentials<VirtualMachine> createNodeWithGroupEncodedIntoName(String group, String name,
+                                                                                    Template template) {
+        logger.info("createNodeWithGroupEncodedIntoName");
+        checkNotNull(template, "template was null");
+        checkNotNull(template.getOptions(), "template options was null");
+        checkArgument(template.getOptions().getClass().isAssignableFrom(AzureTemplateOptions.class),
+                "options class %s should have been assignable from AzureTemplateOptions", template.getOptions().getClass());
+        AzureTemplateOptions templateOptions = template.getOptions().as(AzureTemplateOptions.class);
+        checkNotNull(templateOptions.getStorageAccount(), "storage account should be specified in order to create node");
```

Comment:
This mandatory field needs to be removed. Users must be able to spawn new nodes by just doing:
```java
compute.templageBuilder().build();
```
Consider configuring the storage account as part of the credentials